Implemented double click features for : taxiways, ground vehicules, drapped polygones

IlyassBa opened the pull request.

Commit:
Fix issue with macOS opening the file/folder picker dialog

Commit:
Allow flatten polygons to be submitted to the Gateway

Commit:
Updated internal wed version and switched curl to openssl on Windows

stefanriemens committed:
Fix UB: don't dereference end iterators, they're out-of-bounds!

stefanriemens committed:
Build curl with schannel as ssl on windows. Unfortunately, the curl targets now don't properly link against the necessary libraries. Worked around that by not building the proj executables, and by linking WED against these needed libraries manually.

stefanriemens committed:
Get rid of no-longer-needed openssl dependency on windows.

Commit:
Added a new metadata property to define the altimeter setting for the airport (QNH or QFE)

Commit:
Fix code building for XCode26

Commit:
Updated changedlog

mauer committed:
Merge pull request #38 from X-Plane/wed_270_aiport_meta_data

Added new field in metadate to define the airport altimeter (QNH/QFE)

Commit:
Renamed TYLER_MODE to GATEWAY_IMPORT_MODE

Commit:
Allow headings > 3659 for flatten polygons

Commit:
Adjusted version

IlyassBa committed:
fixed the crashing in the debug. Fix was overloading to use both iterators and pointers in GUI_FitForward(...)

IlyassBa committed:
fixed the crashing in the debug. Fix was overloading to use both iterators and pointers in GUI_FitForward(...)

# Conflicts:
#	src/GUI/GUI_Fonts.cpp

IlyassBa committed:
fixed the crashing in the debug. Fix was overloading to use both iterators and pointers in GUI_FitForward(...)

IlyassBa committed:
fixed the crashing in the debug. Fix was overloading to use both iterators and pointers in GUI_FitForward(...)

stefanriemens committed:
Conan and multi-config generators don't match well, I could not find a way to have all configurations be able to find the dependencies, so instead just generate three separate projects.

stefanriemens committed:
Fix the info.plist, it has to be a dictionary for modern cmake.

IlyassBa committed:
Branch created from master

IlyassBa committed:
Branch created from master

stefanriemens committed:
Add a source-group, so source files are in a directory structure in the generated project file.

stefanriemens committed:
Robots were kind enough to add header files to our targets, so they're part of the generated project files.

stefanriemens committed:
Set CMAKE_OSX_DEPLOYMENT_TARGET in the project.

mauer committed:
Merge pull request #44 from X-Plane/fix_xcode

Fix xcode

Commit:
Fixed flatten polygon export and some general code cleanup

Commit:
Updated UI

mauer committed:
Merge pull request #47 from X-Plane/wed_270_ui

Wed 270 UI

Commit:
Try to fix Workflow for WED

Commit:
Fix syntax

Commit:
Added conan

Commit:
Create conan profile

Commit:
Next try...

Commit:
Install python on Linux

Commit:
Install python on Linux #2

Commit:
Install python on Linux #3

Commit:
Install python on Linux #4

Commit:
Install python on Linux #5

Commit:
Adjust version number

Commit:
Fix issue with icon on macOS

mauer commented:
Hi @IlyassBa,

Can you add a description to the Pull Request, please? I'm not sure which feature you implemented and how it works. In addition, please rebase to wed_270_release.

I tried to compile the topic branch on macOS, but it failed. In this state its unmergable - just like the other Pull Request. It's important that branches compile on Linux, Windows and macOS.

IlyassBa committed:
Merge branch 'feature/double-click-features' of https://github.com/IlyassBa/xptools into feature/double-click-features

IlyassBa committed:
change to initial so if moderator is disable keep the previous code

IlyassBa committed:
Removed Redundant code

IlyassBa commented:
Hi @mauer ,

I rebased from wed_270_release,


The features that i added in here as stated in the name is : 

- OnDoubleClick::Ground Vehicles
- OnDoubleClick::Draped Polygons
- OnDoubleClick::Taxiways


Code wise, its repeated for the three features so i will explain just one and the same applies to all of them.

1. Initial Checks:

```
if (mParent == nullptr) break;
if (gModeratorMode == 0) {
    cell_bounds[0] -= mEditInfo.indent_level * mCellIndent;
    CreateEdit(cell_bounds);
    mClickCellX = cell_x;
    mClickCellY = cell_y;
    return 1;
}
```

=> Here i am checking if no parent table and if moderator mode is disabled.


2. Preparing for Double-Click Logic:

=> I stored the clicked cell coordinates, retrieved the cell's content and toggled the firstDoubleClick  boolean ( Read point 4 about why firstDoubleClick )

3. Double-Click Behavior:

=> If the double clicked cell name is Taxiways i loop all cells (including the ones inside taxiways) to enable the hidden property then i loop inside the Taxiways sub cells and disable the hidden property and then use the DispatchHandleCommand() to enable the SlippyMapESRI


( same for Drapped Polygons and Ground Vehicule )

4. Second Double-Click Behavior: 

This wasn't in the WED Moderator Request Changes PDF, but it makes it easier to go back so if you want me to remove it i just add a comment.

in here I am doing the same thing as in point 3 but i am disabling the hidden property for the the cells.

IlyassBa commented:
Regarding the macOS issue. I currently don’t own a MacBook, but if you can share the macOS error, I’ll be able to get a better understanding of what’s going on
